### DISCUSSION TITLE
RaspberryMatic 3.69.6.20230407

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2278] [Announcements]
This is release 3.69.6.20230407 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.69.6.20230407/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.69.6.20230407) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=78560)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.67.10.20230225...3.69.6.20230407).</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.67.10.20230225...3.69.6.20230404).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.69.6-3 ](https://github.com/jens-maus/occu/tree/b_3_69) integrating full compatibility to the [CCU3 3.69.6 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-69-6.pdf) which comes with the following upstream (eQ3) performed changes:
  - `HmIP-SWD` - Within programs the parameter "Moisture detected: Yes" was wrongly named.
  - Fixed an error when transferring the AutoCal parameter of the `HmIP-SCTH230`.
  - The "CO2 concentration" parameter may not have been displayed correctly within programs.
  - When calling a link of a level channel with a `HmIP-DRBL4` a blank page appeared. blank page.
  - When calling a link of a `HmIP-SWD` with the `HmIP-MP3P` an empty page appeared.
  - Within programs, the percentage value 0 - 100% may have been displayed as 0.0 - 1.0.
  - The configuration parameter `DELAY-COMPENSATION` has been removed for the `HmIP-FROLL` as it is not supported.
  - Correction of the default value for the parameter Condition_Value_HI at `HmIP-BROLL` and `HmIP-BBL`.
  - In certain cases it was possible to upload multiple device firmware updates with different different version status for one device type to the CCU.
  - At the `HmIP-SPI` the calculation of the brightness value was corrected.
  - In the case of wall thermostats with hygrostat function, the currently selected type of two-point control was displayed incorrectly.
  - Some configuration parameters have been added to the `HmIP-eTRV-2` (decalcification run, error position and valve opening degree).
  - The button "Access Points with incompatible FW" (Settings → Control Panel) has been removed.
  - File permissions have been adjusted so that it is now possible to store public keys for SSH access in the file `/root/.ssh/authorized_keys`.
  - For `HmIP-eTRVs` the adaptation drive is now automatically started after a firmware update started.
  - The `INCLUSION_UNSUPPORTED_DEVICE` parameter is no longer offered within programs.
  - With the `HmIP-SPDR`, the channel type is no longer displayed under Status/Operation, but the device name.
  - The `HmIP-HAP` and `HmIPW-DRAP` firmware update files delivered with the CCU were removed.
  - The configuration parameter `CHANNEL_OPERATION_MODE` was added for the `HmIP-BBL` and `HmIP-FBL` with a firmware from 1.10.2.
  - Generierung von SSL Zertifikaten über update.homematic.com wurde aus der WebUI entfernt. Es ist weiterhin möglich über Systemsteuerung → Einstellungen → Netzwerkeinstellungen eigene Zertifikate hoch zu laden oder zu löschen.
  - A new parameter has been added to the Easymode toggle profile for links with the `HmIP-DLD` (Unlock / Open).
  - _Known Bugs_: The Homematic IP DALI gateway does not display a control for 4-channel DALI lights (e.g. RGBW) under Status and operation. Will be fixed in the upcoming version.
- integrated a fixed `HMServer.jar` version which is used in HomeMatic/BidCos-only mode and which will make sure that no homematicIP virtual groups will be unnecessarily removed anymore (#2260).
- updated integrated Mediola NEO server to latest 2.12.2.
- updated `generic_raw_uart` to latest 1.12 version incorporating kernel 6.1.x compatibility.

#### WebUI changes:
- modified `0042-WebUI-HeatingControl-HumidityBatteryValveState` WebUI patch to show "n/a" instead of 20% humidity in heating groups not containing any device that delivers a humidity value (#2274, #2276, @jp112sdl).
- added new `0177-WebUI-Fix-EmptyHmIPWKPPIN` WebUI / OCCU 3.69.6 bugfix patch fixing an incorrect quoting ending up in empty PINs for a HmIP-WKP to show superfluous double apostrophes (#2269).
- modified `0135-WebUI-Add-ControlPanel-AdvancedSettings` WebUI patch to also allow to configure the IP address of a `HB-RF-ETH` adapter directly via the "Advanced settings" WebUI control panel.
- updated CodeMirror to latest 5.65.12.
- modified `0046-WebUI-ImprovedScriptEditor` WebUI patch to allow to resize a CodeMirror based editor vertically. This should enable users to resize the source code and output area in the script editor and script testing popup to their needs. (cf. https://homematic-forum.de/forum/viewtopic.php?p=760691).
- updated `0002-WebUI-Bootstrap` WebUI patch to use a viewport 1024 meta statement to ensure that the WebUI will always appear in desktop mode even on mobile devices because it is not yet ready for mobile devices and does use incorrect scaling since the bootstrap introduction. Also adapted `0036-WebUI-ImprovedFavIcons` accordingly to remove merge conflicts and also adapted `logout/login.htm` to also carry the same meta statements.
- integrated another minor translation typo fix in `0143-WebUI-Fix-Translation` WebUI patch.

#### Operating system changes:
- updated upstream linux kernel for all supported platforms to the latest 6.1.x (6.1.23) LTS version.
- updated RaspberryPi kernel+firmware to latest 1.20230317 version coming with kernel 6.1.x.
- updated buildroot linux environment to latest 2023.02 version.
- modified `install-proxmox.sh` helper script to use "VirtIO SCSI single" controller type in combination with `iothread=1` for improved disk performance in Proxmox. Existing users should make sure to set these parameters in the VM options under Proxmox.
- added a check to `install-proxmox.sh` to output a proper ERROR in case the script is not executed on a Proxmox VE host system (cf. https://homematic-forum.de/forum/viewtopic.php?p=759474#p759447).
- updated tailscale to latest 1.38.3 and modified package to use the official stable packages from pkgs.tailscale.com rather than compiling an own tailscale version from scratch which often comes with go-related compile issues and is better be avoided and should make builds also slightly faster.
- updated wiringpi-rpi package to latest version.
- updated odroid wiringpi package to latest version.
- fixed GPIO name for button use on ODROID-C2.
- added new `multilib32_arm_a55_defconfig` multilib config for ARM cortex A55 cpus and fixed defconfigs for ODROID-C4.
- fixed `HM_LED_GREEN`/`RED` paths for RaspberryPi to use the latest ACT/PWR equivalents in place since kernel 6.1.x/latest Pi firmware. (cf. https://github.com/jens-maus/RaspberryMatic/discussions/2264).
- updated U-boot to latest 2023.01 version for all U-boot enabled platforms (RaspberryPi, ASUS Tinkerboard, ODROID-C4/C2/N2).
- replaced own kernel defconfig for ODROID-C4/C2 with link to same kernel config like ODROID-N2 since these share the same kernel settings anyway.
- reworked `S50lighttpd` to also check the certificate in case the `reload()` function is used.
- updated qemu-guest-agent to latest 7.2.0 version.
- integrated a minor localization fix to `S01InitZRAMSwap` to estimate the total memory amount (cf. https://github.com/StuartIanNaylor/zram-swap-config/commit/eb6fda3a31a0475317a6d871d42a1d5d87d61604)
- modified `S46chrony` startup script to make sure the chrony user can access `/var/lib/chrony` accordingly (#2250).
- enable Microsoft HyperV integration services for ova build target and added upstream patch to fix and optimize the `S10hyperv` init script, thus skip daemon startup if no HyperV environment is found. (cf. https://github.com/home-assistant/operating-system/pull/2381).
- remove unnecessary `0000-Makefile.patch` patch to get `xe-guest-utilities` package compiled again.
- updated `rpi-rf-mod` package to be compatible with latest buildroot check package in version 2023.02.
- updated `xe-guest-utilities` package to latest 7.33.0 to be compatible with buildroot 2023.02.
- added upstream buildroot patch for rtl8812au-aircrack-ng package which updates the package for kernel 6.1.x compatibility.

## :family: Contributors (alphabetically):
- @jens-maus
- @jp112sdl 

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-rpi4.zip)<br/>SHA256: eab2b42add4da62330ad8c4ceed9c1c60c665fe1429ad6c00efb75a17aa0f44a</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-rpi3.zip)<br/>SHA256: ff9042db5157a1f00b62f94dd5727386c1cd7e140eeb35e9e1fb23d455e826ad</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 6286ab3257c760eb0abf65338232bb2320a9aa380dd2db7416bedc5580d16b24</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-rpi2.zip)<br/>SHA256: 1d0a71655a4bd69528665acaf55b211e27b6762d153a86b2b93fe5637d573ee1</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-rpi0.zip)<br/>SHA256: d2f97464a35fc3340cad5fa2508842b628d991876787ed572497d5d227f20b8a</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-tinkerboard.zip)<br/>SHA256: e5f85ea96cdc76ec5775491b3e1120e0ed27640ade77d0c50290785bfb018b05</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-odroid-n2.zip)<br/>SHA256: c581d9f9319f6d0441543f720b9f08cad64951ac606859141fc336912c08296b</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-odroid-c4.zip)<br/>SHA256: 9cf3ba28e7b4b127f4fe744506c8957e4422db15545d1ae626ed00d116028de2</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-odroid-c2.zip)<br/>SHA256: 2e25792a442f77ab18ae6398c972b443962f2a340097dee98fe07fd247bff4d0</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-intelnuc.zip)<br/>SHA256: 06a55c65151a86a6fbccd9cf0cb2f8ff32357fb8287f74a4a0b2bf95b33ff620</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-ova.zip)<br/>SHA256: 9ae8698cebcd6d3c211aae526120441e00cec9df22a31e2183887953dd0c713a</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407.ova) (only for initial OVA installation)<br/>SHA256: 4e2d485b4a2faeadbc2bfdb8910022e5acab29ac4dd690db8aad4cd46ae11ec5</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.69.6.20230407-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 5b0786086831b462dd81921a02339fde5fb6c6d2b62007b74accad921a0647fa</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: ef77b4fe6c56eece4ecdf56e1aed0d74e7aa444af798973338c7f48de5e514fb</sub>
<sub>:package: [RaspberryMatic-3.69.6.20230407-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.6.20230407/RaspberryMatic-3.69.6.20230407-oci_arm.tgz) (arm/armhf)<br/>SHA256: d8575e9b823cbc2897e99441637d9b1328e3cb05f1c9bdbcaad4f529f7e03d26</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.69.6.20230407'>RaspberryMatic 3.69.6.20230407</a>.</em>

**Comment by fastbag** [comment]
4 mal schon über 4 Tage unclean shutdown also absturz
